plugins/services/src/js/utils/DeclinedOffersUtil.js

```diff
@@ -35,6 +36,26 @@ const constraintsReducer = (accumulator, constraint) => {
 };
 
 const DeclinedOffersUtil = {
+  getAppsWithDeclinedOffersFromServiceTree(serviceTree = {}) {
+    if (!serviceTree.list || serviceTree.list.length === 0) {
+      return [];
+    }
+
+    return serviceTree.list.reduce((accumulator, item) => {
+      if (item instanceof ServiceTree) {
+        accumulator = accumulator.concat(
+          DeclinedOffersUtil.getAppsWithDeclinedOffersFromServiceTree(item)
```

Comment:
Please use `Tree.prototype.reduce()` instead of implementing your own recursive function.